dvc/command/diff.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import unicode_literals
+
+import dvc.logger as logger
+from dvc.exceptions import DvcException
+from dvc.command.base import CmdBase
+
+
+class CmdDiff(CmdBase):
+    def _show(self, msg):
+        logger.info(msg)
+
+    def run(self):
+        try:
+            msg = self.repo.diff(
+                self.args.target, a_ref=self.args.a_ref, b_ref=self.args.b_ref
+            )
+            self._show(msg)
+        except DvcException:
+            msg = "failed to get diff "
+            if self.args.target:
```

Comment:
how about this:

`msg += " ".join(compact([target, a.ref, b_ref])) `

where `compact` should be created in the `utils/collections.py` and use `filter` to remove Nones, and empty strings (everything that is bool(s) == False).

Reply:
It's just three ifs is too much for such a simple task :) and too much code complexity, you will have CodeClimate complaining even for this simple wrapper method.